go/api_default.go

```diff
@@ -144,32 +182,296 @@ func getDashboardByName(name string, grafanaUrl string, grafanaApiKey string) (u
+		logrus.Infoln("User not found")
+		return nil
+	}
+
+	logrus.Infoln("User found, deleting...")
+	status, _, err := callGrafana(true, grafanaURL+"/api/admin/users/"+strconv.FormatInt(user.Id, 10), auth, "DELETE", nil)
+
+	if err != nil || status != 200 {
+		return err
+	}
+	return
+}
+
+func addUserToTeam(user User, team Team, grafanaURL string, auth string) (err error) {
+	logrus.Infof("Adding user %s to team %s", user.Email, team.Name)
+
+	member := TeamMember{UserId: user.Id}
+	memberObject, err := json.Marshal(member)
+
+	teamId := strconv.FormatInt(team.TeamId, 10)
+
+	memberPayload := bytes.NewReader(memberObject)
+
+	status, _, err := callGrafana(true, grafanaURL+"/api/teams/"+teamId+"/members", auth, "POST", memberPayload)
+
+	if err != nil {
+		logrus.Infof("Error adding user to team %s", teamId)
+		return
+	}
+
+	if status != 400 && status != 200 {
+		logrus.Infof("Error adding user %s to team %s", strconv.FormatInt(user.Id, 10), teamId)
+		return
+	}
+
+	if status != 400 {
+		logrus.Infoln("User already in team")
+		return
+	}
+
+	if status == 200 {
+		logrus.Infoln("Added user to team")
+		return
+	}
+
+	return
+}
+
+func checkTeamMembership(email string, team Team, grafanaURL string, auth string) (user User, found bool, err error) {
+	logrus.Infof("Checking user %s is in team %s:", email, team.Name)
+
+	teamId := strconv.FormatInt(team.TeamId, 10)
+
+	status, body, err := callGrafana(true, grafanaURL+"/api/teams/"+teamId+"/members", auth, "GET", nil)
+
+	if err != nil || status != 200 {
+		logrus.Infoln("Error checking team membership")
+		return
+	}
+
+	var u User
+
+	if len(body) == 0 {
+		logrus.Infoln("User not found")
+		found = false
+		return u, found, err
+	} else {
+		logrus.Infoln("User found")
+		found = true
+		err := json.Unmarshal(body, &u)
+		return u, found, err
+	}
+}
+
+func getTeamMembers(team Team, grafanaURL string, auth string) (users []User, err error) {
+	logrus.Infof("Getting members of team: %s", team.Name)
+
+	teamId := strconv.FormatInt(team.TeamId, 10)
+
+	status, body, err := callGrafana(true, grafanaURL+"/api/teams/"+teamId+"/members", auth, "GET", nil)
+
+	if err != nil || status != 200 {
+		logrus.Infoln("Error checking team membership")
+		return users, err
+	}
+
+	var members []TeamMember
+	err = json.Unmarshal(body, &members)
+
+	for _, m := range members {
+		user, _, err := getUserByEmail(m.Email, grafanaURL, auth)
+		if err != nil {
+			return users, err
+		}
+		users = append(users, user)
+	}
+
+	if err != nil {
+		logrus.Infoln("Malformed response from Grafana")
+		return users, err
+	}
+	return users, nil
+}
+
+func getTeamByName(name string, grafanaURL string, auth string) (team Team, found bool, err error) {
+	logrus.Infoln("Searching for team by name:", name)
+
+	status, body, err := callGrafana(true, grafanaURL+"/api/teams/search?name="+name, auth, "GET", nil)
+
+	if err != nil {
+		logrus.Infoln("Error getting team with the name:", name)
+		return
+	}
+
+	var g GrafanaGetTeamResponse
```

Comment:
Just extend the `callGrafana()` method and have a out interface{} so you don't need to unmarshall on the outside of each call

Reply:
may have to discuss this one..